### PR TITLE
[results] Implement v1alpha2 Create/GetResult.

### PR DESCRIPTION
**Pull request by wlynch**
# Changes

Implements the CreateResult and GetResult methods of the v1alpha2
interface.

Includes request validation, and some basic error handling.
Unfortunately since SQL error codes are somewhat impl specific, we will
need to do some additional work to fill out error handling (particularly
for MySQL). For now, focus on gorm/sqlite errors for unit testing.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [n/a] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Commit messages includes a project tag in the subject - e.g. [OCI], [hub], [results], [task-loops]

_See [the contribution guide](https://github.com/tektoncd/experimental/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: results/pkg/api/server/test/db.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
+package test
```
Comment (dibyom) [inline results/pkg/api/server/test/db.go:1]: In other tekton projects, we've been following the pattern of putting all test helpers/e2e tests in a separate test/ folder that is in the same level as pkg/

**Review by dibyom — COMMENTED** [review]
/approve

File: results/pkg/api/server/test/db.go (hunk 1 above)
Comment (wlynch) [reply results/pkg/api/server/test/db.go:1]: I'm not sure we want to do that here - I suspect that we may want to split the API Server / watcher into separate modules in the future, so keeping this closer to the rest of the API Server code will make that transition easier if/when that happens.

**Pull request merged**